oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java

```diff
@@ -778,7 +778,7 @@ private <T extends Document> List<T> queryWithRetry(Collection<T> collection,
             }
         }
         if (ex != null) {
-            throw handleException(ex, collection, Lists.newArrayList(fromKey, toKey));
+            throw handleException(ex, collection, List.of(fromKey, toKey));
```

Comment:
It's at least not obvious that fromKey and toKey cannot be null.

Reply:
If you look at the caller and the callee it's reasonable to assume...

Reply:
In this particular case, but there might be other callers in the future.

Reply:
Then those would fail.